dotnet/src/Experimental/Process.Core/ProcessStepBuilder.cs

```diff
@@ -46,6 +46,16 @@ public virtual ProcessStepEdgeBuilder OnFunctionResult(string functionName)
         return this.OnEvent($"{functionName}.OnResult");
     }
 
+    /// <summary>
+    /// Define the behavior of the step when the specified function has thrown an exception.
+    /// </summary>
+    /// <param name="functionName">The name of the function of interest.</param>
+    /// <returns>An instance of <see cref="ProcessStepEdgeBuilder"/>.</returns>
+    public ProcessStepEdgeBuilder OnFunctionError(string functionName)
```

Comment:
perhaps one of the existing samples of `GettingStartedWithProcesses` should showcase this.

Maybe one of the cooking samples -> FryFoodStep -> fryerMalfunction  if range 0-4 throw exception 4-6 emit FoodRuinded, else success.